quinn-proto/src/connection/assembler.rs

```diff
@@ -29,6 +29,16 @@ impl Assembler {
         Self::default()
     }
 
+    /// Reset to the initial state
+    pub(super) fn reinit(&mut self) {
+        self.state = State::default();
+        self.data.clear();
+        self.buffered = 0;
+        self.allocated = 0;
+        self.bytes_read = 0;
+        self.end = 0;
```

Comment:
Could this be simplified to something like the below?
```rs
let old = mem::take(self);
self.data = old.data;
self.data.clear();
```
An empty `BinaryHeap` shouldn't allocate, and will likely get optimized out entirely. That way we avoid the hazard of forgetting to update this with new fields.

Reply:
done